packages/harmony/src/components/button/Button/Button.mdx

```diff
@@ -0,0 +1,134 @@
+import {
+  Canvas,
+  Meta,
+  Title,
+  Subtitle,
+  Description,
+  Primary,
+  Controls,
+  Story,
+  Stories
+} from '@storybook/blocks'
+import { Button } from './Button'
+import { ButtonSize, ButtonType } from '../types'
+import { Flex, Box } from 'components'
+import * as ButtonStories from './Button.stories'
+import test from '../../../../.storybook/public/buttonBackground.png'
+import IconCamera from 'assets/icons/Camera.svg'
+
+<Meta of={ButtonStories} />
+
+<Title />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Variants](#variants)
+- [Use cases and examples](#use-cases-and-examples)
+- [Do's and dont's](#dos-and-donts)
+
+## Overview
+
+<Subtitle />
+<Description />
+<Primary />
+
+## Props
+
+<Controls />
+## Variants
+
+<Canvas>
+  <Story of={ButtonStories.ButtonTypes} />
+</Canvas>
+
+<Stories />
+
+## Use cases and examples
+
+- Always use primary button as the main action on screen.
+- Avoid vague or generic labels that requires the user read before taking action
+- Use active verbs or phrases that clearly indicate action.
+
+## Do's and don'ts
+
+<ComponentRules
+  rules={[
```

Comment:
Thanks I am considering to use a component hierarchy instead of config object, lmk what you think

Reply:
Like this? 
```jsx
<ComponentRules>
	<Do description=''>
		<Button />
	</Do>
	<Dont description=''>
		<Button />
	</Dont>
</ComponentRules>
```

I think either is fine. Maybe the component hierarchy is more flexible whereas the config file keeps the structure in the parent. I kinda like the config because it lets us change the structure in one place if we ever want to reorganize the display.